Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/overview/PersonOverviewFragment.kt

```diff
@@ -1,4 +1,107 @@
 package de.rki.coronawarnapp.covidcertificate.person.ui.overview
 
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.DefaultItemAnimator
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.covidcertificate.person.ui.overview.items.CertificatesItem
+import de.rki.coronawarnapp.databinding.PersonOverviewFragmentBinding
+import de.rki.coronawarnapp.util.DialogHelper
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.lists.decorations.TopBottomPaddingDecorator
+import de.rki.coronawarnapp.util.lists.diffutil.update
+import de.rki.coronawarnapp.util.onScroll
+import de.rki.coronawarnapp.util.ui.doNavigate
+import de.rki.coronawarnapp.util.ui.viewBinding
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import javax.inject.Inject
+
 // Shows a list of multiple persons
-class PersonOverviewFragment
+class PersonOverviewFragment : Fragment(R.layout.person_overview_fragment), AutoInject {
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: PersonOverviewViewModel by cwaViewModels { viewModelFactory }
+    private val binding by viewBinding<PersonOverviewFragmentBinding>()
+    private val personOverviewAdapter = PersonOverviewAdapter()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        binding.apply {
+            bindToolbar()
+            bindRecycler()
+            scanQrcodeFab.setOnClickListener { viewModel.onScanQrCode() }
+        }
+        viewModel.personCertificates.observe(viewLifecycleOwner) { binding.bindViews(it) }
+        viewModel.events.observe(viewLifecycleOwner) { onNavEvent(it) }
+    }
+
+    private fun onNavEvent(event: PersonOverviewFragmentEvents) {
+        when (event) {
+            is OpenPersonDetailsFragment -> doNavigate(
+                PersonOverviewFragmentDirections.actionPersonOverviewFragmentToVaccinationListFragment(
+                    event.personIdentifier
+                )
+            )
+            is ShowDeleteDialog -> {
+                val dialog = DialogHelper.DialogInstance(
```

Comment:
I would prefer we stop using the `DialogHelper`, I don't think it really helps (at least in its current form), vs just using `MaterialAlertDialogBuilder`.